test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala

```diff
@@ -97,4 +97,12 @@ class CharsetTest {
             "iso-ir-100", "latin1", "cp819", "ISO8859_1", "IBM819", "ISO_8859_1",
             "IBM-819", "csISOLatin1"))
   }
+
+  @Test def availableCharsets(): Unit = {
+    val c = Charset.availableCharsets()
+
+    assertSame(US_ASCII, c.get("us-ASCII"))
```

Comment:
Some more test ideas:
* Test the other supported ones, for completeness?
* Test that we cannot find charsets by their *aliases* this way
* Test that we cannot find any non-existing charset either
* Actually test the *listing* of all the expected charsets (in order) -- this needs to be guarded by an `!isOnJVM` or something